saq/queue.py

```diff
@@ -328,31 +336,35 @@ async def listen(
             callback: callback function, if it returns truthy, break
             timeout: if timeout is truthy, wait for timeout seconds
         """
-        pubsub = self.redis.pubsub()
+        if not job_keys:
```

Comment:
why do you need this check? you can't check if job_keys is empty because it's an iterable

Reply:
whoops, good catch.

Doesn't really make sense to PubSubMultiplexer.subscribe().
I'll just move the check to job_ids.